src/frontend/src/features/auth/cognito-auth.tsx

```diff
@@ -0,0 +1,312 @@
+import React, {createContext, useContext, useEffect, useState} from 'react';
+import {Auth, Amplify} from 'aws-amplify';
+
+
+const AwsConfigAuth = {
+  region: import.meta.env.VITE_COGNITO_REGION,
+  userPoolId: import.meta.env.VITE_COGNITO_USER_POOL_ID,
+  userPoolWebClientId: import.meta.env.VITE_COGNITO_WEB_CLIENT_ID,
+  authenticationFlowType: 'USER_SRP_AUTH',
+}
+
+Amplify.configure({Auth: AwsConfigAuth});
+
+interface UseAuth {
+  isLoading: boolean;
+  isAuthenticated: boolean;
+  username: string;
+  currentAuthenticatedUser: () => Promise<any>;
+  signUp: (username: string, password: string) => Promise<Result>;
+  confirmSignUp: (verificationCode: string) => Promise<Result>;
+  signIn: (username: string, password: string) => Promise<Result>;
+  signInComplete: (username: string, oldPassword: string, newPassword: string) => Promise<Result>;
+  signOut: () => void;
+  changePassword: (user: any, oldPassword: string, newPassword: string, newPasswordConfirm: string) => Promise<Result>;
+  forgetPassword: (email: string) => Promise<Result>;
+  resetPassword: (username: string, code: string, newPassword: string, newPasswordConfirm: string) => Promise<Result>;
+}
+
+interface Result {
+  success: boolean
+  message: string
+  hasChallenge?: boolean
+  challengeName?: string
+}
+
+const authContext = createContext({} as UseAuth);
+
+export const ProvideAuth: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const auth = useProvideAuth();
+  return <authContext.Provider value={auth}>{children}</authContext.Provider>;
+};
+
```

Comment:
providerはfeaturesで書かない。

featuresの外にproviders/を作ってそこに
```suggestion
export const ProvideAuth: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const auth = useProvideAuth();
  return <authContext.Provider value={auth}>{children}</authContext.Provider>;
};

```